grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/plan/priority/ResourcePriority.java

```diff
@@ -0,0 +1,102 @@
+ */
+
+package ai.grakn.graql.internal.reasoner.plan.priority;
+
+import ai.grakn.graql.Var;
+import ai.grakn.graql.admin.VarPatternAdmin;
+import ai.grakn.graql.internal.reasoner.atom.Atom;
+import ai.grakn.graql.internal.reasoner.atom.predicate.ValuePredicate;
+import ai.grakn.graql.internal.reasoner.plan.SimplePlanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * <p>
+ * Class defining resolution weight for resource atoms.
+ * </p>
+ *
+ * @author Kasper Piskorski
+ *
+ */
+class ResourcePriority extends BasePriority {
+
+    ResourcePriority(Atom atom) {
+        super(atom);
+    }
+
+    private boolean isSuperNode(){
+        return atom().getParentQuery().tx().graql().match(atom().getCombinedPattern()).admin().stream()
+                .skip(SimplePlanner.RESOURCE_SUPERNODE_SIZE)
+                .findFirst().isPresent();
+    }
+
+    @Override
+    public int computePriority(Set<Var> subbedVars){
+        int priority = super.computePriority(subbedVars);
+        Set<ai.grakn.graql.ValuePredicate> vps = atom().getPredicates(ValuePredicate.class).map(ValuePredicate::getPredicate).collect(Collectors.toSet());
+        priority += SimplePlanner.IS_RESOURCE_ATOM;
+
+        if (vps.isEmpty()) {
+            if (subbedVars.contains(atom().getVarName()) || subbedVars.contains(atom().getPredicateVariable())
+                    && !isSuperNode()) {
+                priority += SimplePlanner.SPECIFIC_VALUE_PREDICATE;
+            } else{
+                priority += SimplePlanner.VARIABLE_VALUE_PREDICATE;
+            }
+        } else {
+            int vpsPriority = 0;
+            for (ai.grakn.graql.ValuePredicate vp : vps) {
+                //vp with a value
+                if (vp.isSpecific() && !isSuperNode()) {
+                    vpsPriority += SimplePlanner.SPECIFIC_VALUE_PREDICATE;
+                } //vp with a variable
+                else if (vp.getInnerVar().isPresent()) {
+                    VarPatternAdmin inner = vp.getInnerVar().orElse(null);
+                    //variable mapped inside the query
+                    if (subbedVars.contains(atom().getVarName())
+                            || subbedVars.contains(inner.var())
+                            && !isSuperNode()) {
+                        vpsPriority += SimplePlanner.SPECIFIC_VALUE_PREDICATE;
+                    } //variable equality
+                    else if (vp.equalsValue().isPresent()){
+                        vpsPriority += SimplePlanner.VARIABLE_VALUE_PREDICATE;
+                    } //variable inequality
+                    else {
+                        vpsPriority += SimplePlanner.COMPARISON_VARIABLE_VALUE_PREDICATE;
+                    }
+                } else {
+                    vpsPriority += SimplePlanner.NON_SPECIFIC_VALUE_PREDICATE;
+                }
+            }
+            //normalise
+            vpsPriority = vpsPriority/vps.size();
```

Comment:
Is it accurate enough to use integer here? I'm asking because I don't really know how big this number is.